disnake/embeds.py

```diff
@@ -821,9 +886,15 @@ def get_default_colour(cls) -> Optional[Colour]:
 
     get_default_color = get_default_colour
 
-    def _handle_resource(self, url: Optional[Any], file: File, *, key: _FileKey) -> Optional[str]:
-        if not (url is MISSING) ^ (file is MISSING):
-            raise TypeError("Exactly one of url or file must be provided")
+    def _handle_resource(
+        self, url: Optional[Any], file: File, *, key: _FileKey, required: bool = True
```

Comment:
```suggestion
        self, url: Optional[Any], file: Optional[File], *, key: _FileKey, required: bool = True
```
tiny non-runtime nit, pyright considers the `else` branch unreachable otherwise